tests/Spice86.Tests/Emulator/Devices/Sound/SoundBlasterDspPortTests.cs

```diff
@@ -0,0 +1,455 @@
+
+        context.FreqHz.Should().Be(8000);
+    }
+
+    [Fact]
+    public void TestRegisterStoresLastValue() {
+        using SoundBlasterPortTestContext context = new SoundBlasterPortTestContext();
+
+        context.SendDspCommand(0xE4, 0x5A);
+
+        context.DspTestRegister.Should().Be(0x5A);
+    }
+
+    private sealed class SoundBlasterPortTestContext : IDisposable {
+        private readonly object _irqState;
+        private readonly PropertyInfo _pending8BitProperty;
+        private readonly PropertyInfo _pending16BitProperty;
+        private readonly PropertyInfo _speakerEnabledProperty;
+        private readonly PropertyInfo _freqHzProperty;
+        private readonly PropertyInfo _dspStateProperty;
+        private readonly PropertyInfo _dspTestRegisterProperty;
+        private readonly object _soundBlasterState;
+        private readonly object _dspStateInstance;
+
+        public SoundBlasterPortTestContext(SbType sbType = SbType.SBPro2) {
+            ILoggerService loggerService = Substitute.For<ILoggerService>();
+            AddressReadWriteBreakpoints memoryBreakpoints = new();
+            AddressReadWriteBreakpoints ioBreakpoints = new();
+            State state = new(CpuModel.INTEL_80286);
+            IOPortDispatcher dispatcher = new(ioBreakpoints, state, loggerService, failOnUnhandledPort: false);
+
+            Ram ram = new(A20Gate.EndOfHighMemoryArea);
+            A20Gate a20Gate = new();
+            Memory memory = new(memoryBreakpoints, ram, a20Gate, initializeResetVector: false);
+
+            DmaBus dmaBus = new(memory, state, dispatcher, failOnUnhandledPort: false, loggerService);
+            DualPic dualPic = new(dispatcher, state, loggerService, failOnUnhandledPort: false);
+            IEmulatedClock clock = new EmulatedClock();
+            EmulationLoopScheduler scheduler = new(clock, loggerService);
+            Mixer mixer = new(loggerService, AudioEngine.Dummy);
+
+            HashSet<ChannelFeature> oplFeatures = new HashSet<ChannelFeature> {
+                ChannelFeature.Synthesizer,
+                ChannelFeature.Sleep,
+                ChannelFeature.Stereo
+            };
+            MixerChannel oplChannel = mixer.AddChannel(_ => { }, 49716, "OplTestChannel", oplFeatures);
+
+            SoundBlasterHardwareConfig config = new SoundBlasterHardwareConfig(7, 1, 5, sbType);
+            SoundBlaster soundBlaster = new SoundBlaster(
+                dispatcher,
+                state,
+                dmaBus,
+                dualPic,
+                mixer,
+                oplChannel,
+                loggerService,
+                scheduler,
+                clock,
+                config);
+
+            Dispatcher = dispatcher;
+            DualPic = dualPic;
+            Mixer = mixer;
+            Config = config;
+            SoundBlaster = soundBlaster;
+
+            FieldInfo sbField = typeof(SoundBlaster).GetField("_sb", BindingFlags.NonPublic | BindingFlags.Instance)
```

Comment:
don't use reflection, add public properties to the SB.

same for all classes that are put under test = don't use reflection.